torchsynth/synth.py

```diff
@@ -125,12 +126,14 @@ def randomize(self, seed: Optional[int] = None):
             mt19937_gen = csprng.create_mt19937_generator(seed)
             for parameter in self.parameters():
                 parameter.data.uniform_(0, 1, generator=mt19937_gen)
-            for module in self._modules:
-                self._modules[module].seed = seed
         else:
             for parameter in self.parameters():
                 parameter.data = torch.rand_like(parameter, device=self.device)
 
+        # Add see to all modules
+        for module in self._modules:
+            self._modules[module].seed = seed
```

Comment:
Seed should always be set -- if randomize was previously called with a seed then that seed will remain in the modules unless we replace it with None when calling randomize without a seed